lsp-mode.el

```diff
@@ -4440,16 +4440,21 @@ Also, additional data to attached to each candidate can be passed via PLIST."
                          "insertTextFormat" insert-text-format
                          "additionalTextEdits" additional-text-edits)
                   item))
-           (cond
-            (text-edit
-             (delete-region (point-at-bol) (point))
-             (insert prefix-line)
-             (lsp--apply-text-edit text-edit))
-            ((or insert-text label)
-             (delete-region (point-at-bol) (point))
-             (insert prefix-line)
-             (delete-region start-point (point))
-             (insert (or insert-text label))))
+           (let ((bol (point-at-bol)))
```

Comment:
will that work if I do: 

type `f` | invoke completion | type `oo` | go back one char | auto-complete `foo()`

Reply:
I think the behavior will not change compare to before.
Before: we delete the region from `bol` to `(point)` and insert `prefix-line` back to it.
Now: we compare if the string from `bol` to `(point)` already include `prefix-line` and only delete the different part.

Both of them should resulted in same text, only different is that the later is changing smaller region than the former

Reply:
It does not. The second one will result - `fooo()` because the `o` will be omitted from `(delete-region (+ bol (length prefix-line)) (point))`

Reply:
Apparently this does not work even now... we could fix it probably by keeping suffix-line(or using markers?) or replace-bufffer-contents with the full line.

Reply:
The first one is not? I think it will resulted in the same thing, unless we include the `suffix-line` to also restore behind the point

Reply:
For snippets I am used to it works well.
Should it be considered to remove `(setq-local yas-inhibit-overlay-modification-protection t)` ?

Reply:
Using marker seems nice, let's me try it.

Reply:
Switch to use markers instead of `prefix-line`.